include/dlaf/matrix/print_numpy.h

```diff
@@ -0,0 +1,117 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2020, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <string>
+
+#include "dlaf/common/index2d.h"
+#include "dlaf/common/range2d.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/matrix/tile.h"
+#include "dlaf/types.h"
+
+namespace dlaf {
+namespace matrix {
+
+namespace internal {
+
+// For reference see https://numpy.org/doc/stable/user/basics.types.html
+// Moreover, considering the small number of digits in the output,
+// just single-precision types are used.
+template <class T>
+struct numpy_datatype {
+  static constexpr auto typestring = "single";
+};
+
+template <class T>
+struct numpy_datatype<std::complex<T>> {
+  static constexpr auto typestring = "csingle";
+};
+
+template <class T>
+std::string numpy_value(const T& value) {
+  std::ostringstream os;
+  os << value;
+  return os.str();
+}
+
+template <class T>
+std::string numpy_value(const std::complex<T>& value) {
+  std::ostringstream os;
+  const auto imag = std::imag(value);
+  const auto sign = (imag < 0) ? '-' : '+';
+  os << std::real(value) << sign << std::abs(imag) << "j";
+  return os.str();
+}
+}
+
+/// Print a tile as a numpy array
+template <class Stream, class T>
+Stream& print_numpy(Stream& os, const Tile<const T, Device::CPU>& tile) {
```

Comment:
Is there a reason to use a template instead of `std::ostream`?

Reply:
No, it was a leaked experiment. 😁